code/modules/holiday/holidays.dm

```diff
@@ -30,15 +30,28 @@
 	/// The default pattern of the holiday, if the requested pattern is null.
 	var/holiday_pattern = PATTERN_DEFAULT
 
+/datum/holiday/New()
+	for(var/timezone in timezones)
+		var/time_in_timezone = world.realtime + timezone HOURS
+
+		var/YYYY = text2num(time2text(time_in_timezone, "YYYY")) // get the current year
+		var/MM = text2num(time2text(time_in_timezone, "MM")) // get the current month
+		var/DD = text2num(time2text(time_in_timezone, "DD")) // get the current day
+		var/DDD = time2text(time_in_timezone, "DDD") // get the current weekday
+
+		if(shouldCelebrate(DD, MM, YYYY, DDD))
```

Comment:
It's not a huge deal but it seems unfortunate to go from our previous system where we get all of these dates a single time to recalculating today's date once for every holiday, can you just pass the date info into `New`?

Reply:
Oh you are right, you can ignore this then.